### PR TITLE
Readability of tween_interpolaters.cpp improved, fixes #21600

### LINKED ISSUE #21600: Improve readability on math intensive code, like tween_interpolaters.cpp
**Godot version:**
3.1 (master)

**OS/device including version:**
Windows 10

**Issue description:**
There are some files in the cpp codebase that look like obfuscated code.
Example:
image

**Steps to reproduce:**
1. Open tween_interpolaters.cpp
2. Try to understand/modify/create derivatives of that code
3. Fail because of obfuscated-like code

I understand this is subjective, but there are still some improvements like e.g. using readable variable names, that can improve community collaboration. BTW I have no idea what those variables are.

### PR DESCRIPTION
**Pull request by jahd2602**
fixes #21600

### CODE REVIEW

File: scene/animation/tween_interpolaters.cpp [hunk 1]
```diff
@@ -36,277 +36,277 @@ const real_t pi = 3.1415926535898;
 // linear
 ///////////////////////////////////////////////////////////////////////////
 namespace linear {
-static real_t in(real_t t, real_t b, real_t c, real_t d) {
-	return c * t / d + b;
+static real_t in(real_t time, real_t beginning, real_t change, real_t duration) {
+    return change * time / duration + beginning;
```
Comment (LikeLakers2) [inline scene/animation/tween_interpolaters.cpp]: Indentation should be tab-based, not space-based. http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

**Comment by jahd2602** [comment]
@LikeLakers2 I just amended my commit.

File: scene/animation/tween_interpolaters.cpp [hunk 2]
```diff
@@ -36,277 +36,277 @@ const real_t pi = 3.1415926535898;
...
+	if ((time /= duration / 2) == 2) return beginning + change;
+	float p = duration * (0.3f * 1.5f);
+	float a = change;
 	float s = p / 4;
 
-	if (t < 1) {
-		float postFix = a * pow(2, 10 * (t -= 1)); // postIncrement is evil
-		return -0.5f * (postFix * sin((t * d - s) * (2 * pi) / p)) + b;
+	if (time < 1) {
+		float postFix = a * pow(2, 10 * (time -= 1)); // postIncrement is evil
```
Comment (akien-mga) [inline scene/animation/tween_interpolaters.cpp]: The whole point of the "improve readability" was to get rid of these inline operations like `time -= 1` or `time /= duration`.

Replacing all variable names without changing that is IMO pointless, it's not more readable with long variables names than with well documented short variable names (they have the same meaning in all methods, so t, b, c d is fine by me - what's not is these too clever one liners).

**Review by akien-mga — CHANGES_REQUESTED** [review]

File: scene/animation/tween_interpolaters.cpp (hunk 2 above)
Comment (jahd2602) [reply scene/animation/tween_interpolaters.cpp]: The first and mandatory step was to change the names of the variables. 
Getting rid of these inline operations may be beneficial too.
But at least, this PR covers the first step.

**Comment by akien-mga** [comment]
Since this is thirdparty code, I prefer that we don't modify it if we can avoid it, especially if it's only cosmetic. There are actual bugs in this file due to undefined behaviour with inline variable assignments (see https://github.com/godotengine/godot/pull/22446#issuecomment-425058229).

Given the poor state this code is in, we should find a properly maintained replacement (the project the code was taken from has very bad code quality and no improvements since 2008). In the meantime, I'm moving it to our `thirdparty/` folder in #22512 to make it clear that it's not Godot code and we are not its maintainers (and disable warnings when building it).

**Pull request closed without merging**